### PR TITLE
WIP: Made more generic to implement 1.14

### LINKED ISSUE #5: Shield slot not handled
When using the clickWindow function. You can select items and place them everywhere except for
the shield slot. When trying to click the shield slot, it will work (i think) but not update the information for your bot. So you get no response, the bot thinks it dident click on the slot, wich will cause errors if you call clickWindow again. (and drop your item)

The shield slot is the last slot in the inventory. To fix this increase [link](https://github.com/PrismarineJS/prismarine-windows/blob/eece07c2697200888c14b0bd3308b31c99c63f5a/lib/windows.js#L30)
> var INVENTORY_SLOT_COUNT = 36;

by 1, to 
> var INVENTORY_SLOT_COUNT = 37;

I am again not sure if this breaks something else, so far it works fine for me. If it causes any problem i will come back and report.

### PR DESCRIPTION
**Pull request by Karang**
Needs review and testing

Closes #5

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,6 +1,86 @@
 function loader (mcVersion) {
+  const Item = require('prismarine-item')(mcVersion)
+  const Window = require('./lib/Window')(Item)
+
+  let windows
+  if (mcVersion.startsWith('1.14')) {
```
Comment (rom1504) [inline index.js:6]: would be better to retrieve majorVersion from mcData, this check doesn't work for snapshot

File: index.js [hunk 2]
```diff
@@ -1,6 +1,86 @@
...
+    // https://wiki.vg/Inventory
+    windows = {
+      'minecraft:inventory': { type: -1, inventory: { start: 9, end: 44 }, slots: 46, craft: 0, requireConfirmation: true },
```
Comment (rom1504) [inline index.js:9]: very nice, would be nice if we could put this in mcdata (related to current windows.json I guess ?)

File: index.js (hunk 1 above)
Comment (Karang) [reply index.js:6]: The real fix would be to put the json in mcData and get the one for the right version.

File: index.js (hunk 2 above)
Comment (Karang) [reply index.js:9]: yes, is the current windows.json actually used anywhere? Can we replace it or do we merge or create another json?

**Comment by rom1504** [comment]
It was initially added by @Gjum. Do you still use it @gjum ?

Maybe we could have a look at the format to see if it can store all the info we need or not

**Comment by Gjum** [comment]
I am using it [here](https://github.com/Gjum/jmcdata/blob/master/src/main/kotlin/com/github/gjum/minecraft/jmcdata/Windows.kt) with 1.12 and eventually 1.14+15

it's also used in [spockbot](https://github.com/SpockBotMC/SpockBot/blob/master/spockbot/mcdata/windows.py#L4)

what changes to the format would you want to do? I haven't looked into how guis work in 1.13+ yet. The data structure in this PR looks like it can be represented by the existing format though.

**Comment by rom1504** [comment]
Okay, we will find a way to fit our information into this format then,
hopefully it's possible

**Comment by rom1504** [comment]
@Karang this was a breaking change right ?
I didn't notice and published it as a minor version. completely breaking flying-squid

**Comment by rom1504** [comment]
too late to do anything at this point, mineflayer was already released depending on this version, looks like only mineflayer and flying-squid depend on it, so I'll just update flying-squid

**Pull request merged**